File: Algo/Leetcode/083RemoveDuplicatesFromSortedList.cpp
//Use two pointer. Prev pointer update to cur if finally go out the equal ones.
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(!head) return head;
        ListNode* move = head;
        while(move->next){
            if(move->next->val == move->val){
                move->next = move->next->next;
            } else move=move->next;
        }
        return head;
    }
};

File: Algo/Leetcode/083RemoveDuplicatesFromSortedList.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head is None:
            return None

        m = head
        while m and m.next and m.val == m.next.val:
            m = m.next

        if m.next is None:
            head.next = None
        else:
            head.next = self.deleteDuplicates(m.next)

        return head